BookStore/controllers/user.js

```diff
@@ -32,4 +32,27 @@ exports.update = (req, res) => {
             req.user.salt = undefined;
             res.json(user);
         })
-}
\ No newline at end of file
+}
+
+exports.addOrderToUserHistory = (req, res, next) => {
+    let history = [];
+    req.body.order.products.forEach((item) => {
+        history.push({
+            _id: item._id,
+            name: item.name,
+            description: item.description,
+            category: item.category,
+            quantity: item.quantity,
+            transaction_id: req.body.order.transaction_id,
+            amount: req.body.amount
+        });
+    })
+    User.findOneAndUpdate({ _id: req.user._id },
+        { $push: { history: history } },
```

Comment:
You could use [ES6 object initialiser syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Syntax) here to make this neater. Eg:

```js
{ $push: { history } },
```

Reply:
done